tools/topology/sof-cnl-nocodec.m4

```diff
@@ -0,0 +1,69 @@
+#
+# Topology for generic Cannonlake board with nocodec
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`ssp.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Cannonlake DSP configuration
+include(`platform/intel/cnl.m4')
+
+#
+# Define the pipelines
+#
+# PCM0 <---> Volume <---> SSP0 (NoCodec)
+#
+
+# Low Latency playback pipeline 1 on PCM 0 using max 2 channels of s24le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_DAI_ADD(sof/pipe-volume-playback.m4,
+	1, 0, 2, s24le,
+	48, 1000, 0, 0, SSP, 0, s24le, 2)
+
+# Low Latency capture pipeline 2 on PCM 0 using max 2 channels of s24le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_DAI_ADD(sof/pipe-volume-capture.m4,
+	2, 0, 2, s24le,
+	48, 1000, 0, 0, SSP, 0, s24le, 2)
```

Comment:
The pinmux of WHL shows there are up to 3 SSPs, why use SSP0 only?
Also you have a 2 DMIC interfaces, why not allow for them?

Reply:
@plbossart OK, I will add them all, the first purpose is to make the RVP run with FW, so keep it simple at the first time.

Reply:
@plbossart Have add 3 SSPs and one DMIC